config/routes.rb

```diff
@@ -6,5 +6,11 @@
   get "up" => "rails/health#show", as: :rails_health_check
 
   # Defines the root path route ("/")
-  # root "posts#index"
+  # Set root to home#index for the base URL
+  root 'home#index'
+
+  # Individual routes for users, their posts, and specific posts
+  get 'users/:user_id', to: 'users#show', as: 'user'
+  get 'users/:user_id/posts', to: 'posts#index', as: 'user_posts'
+  get 'users/:user_id/posts/:id', to: 'posts#show', as: 'user_post'
 end
```

Comment:
- [ ] Please ensure that you do not expose unnnecessary endpoints by using the `only` keyword for your routes

e.g
```



# config/routes.rb
  root 'home#index
  resources :users, only: [:index, :show] do
    resources :posts, only: [:index, :show]
  end

```